### PR TITLE
AG-8218 Add initial zoom options

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-8218

- Adds initial zoom options with priority over navigator, and changes zoom to use the `StateTracker` instead of the `initialZoom` and `pendingZoom` values.
- Renames `state-tracker.ts` to `stateTracker.ts`
- Fixes a duplication in validation messages for ratios of `expecting expected ...`

I'm going to extract the zoom properties out into a new class, but didn't want to muddy this change.

### CODE REVIEW

File: packages/ag-charts-community/src/chart/interaction/zoomManager.ts [hunk 1]
```diff
@@ -118,8 +124,8 @@ export class ZoomManager extends BaseManager<'zoom-change', ZoomChangeEvent> {
 class AxisZoomManager {
     private axis: ChartAxisLike;
     private currentZoom: ZoomState;
-    private pendingZoom: ZoomState | null = null;
     private readonly initialZoom: ZoomState;
+    private state = new StateTracker<ZoomState>();
```
Comment (iMoses) [inline packages/ag-charts-community/src/chart/interaction/zoomManager.ts]: nit: if initialised inside the constructor, you can remove `initialZoom` by providing the value as the default state value.
```
// replace:
this.initialZoom = this.currentZoom = { min, max };
// with:
this.state = new StateTracker({ min, max });
this.currentZoom = this.state.stateValue();
```

File: packages/ag-charts-community/src/chart/interaction/zoomManager.ts [hunk 2]
```diff
@@ -40,27 +41,32 @@ export class ZoomManager extends BaseManager<'zoom-change', ZoomChangeEvent> {
             this.axisZoomManagers.set(axis.id, zoomManagers.get(axis.id) ?? new AxisZoomManager(axis));
         }
 
-        if (this.initialZoom) {
-            this.updateZoom(this.initialZoom);
-            this.initialZoom = undefined;
+        if (this.state.size > 0) {
+            this.updateZoom(this.state.stateId() ?? 'initial', this.state.stateValue());
```
Comment (iMoses) [inline packages/ag-charts-community/src/chart/interaction/zoomManager.ts]: nit: you can also specify a default id
```
private state = new StateTracker<AxisZoomState>(null, 'initial');
```

File: packages/ag-charts-community/src/chart/interaction/zoomManager.ts (hunk 1 above)
Comment (lsjroberts) [reply packages/ag-charts-community/src/chart/interaction/zoomManager.ts]: Oh of course, thanks!

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**